Ical.Net.Tests/AlarmTest.cs

```diff
@@ -91,4 +94,33 @@ public void Remove1Attendee()
         evt.Attendees.Remove(_attendees.Last());
         Assert.That(evt.Attendees.Count, Is.EqualTo(0));
     }
+
+    [Test, Category("Alarm Trigger")]
+    [TestCase("PT1H", 0, 0, 1, 0, 0)]
+    [TestCase("PT1H30M", 0, 0, 1, 30, 0)]
+    [TestCase("PT1H30M45S", 0, 0, 1, 30, 45)]
+    [TestCase("P1D", 0, 1, 0, 0, 0)]
+    [TestCase("P1W", 1, 0, 0, 0, 0)]
+    [TestCase("-P1D", 0, -1, 0, 0, 0)]
+    [TestCase("-P1W", -1, 0, 0, 0, 0)]
+    [TestCase("PT0S", 0, 0, 0, 0, 0)]
+    [TestCase("-PT1H", 0, 0, -1, 0, 0)]
+    [TestCase("-PT1H30M", 0, 0, -1, -30, 0)]
+    [TestCase("-P1D", 0, -1, 0, 0, 0)]
+    [TestCase("PT25H99M99S", 0, 0, 25, 99, 99)]
+    [TestCase("-P0DT0H30M0S", 0, 0, 0, -30, 0)]
+    public void AlarmTrigger_ShouldReturn_ExpectedDuration(string text, int? weeks = null, int? days = null, int? hours = null, int? minutes = null, int? seconds = null)
```

Comment:
Naming/class: This test is related to `Duration`s, not to `Alarm`.

Reply:
Absolutely